### DISCUSSION TITLE
Progressive non-binary marks for check list?

### DISCUSSION THREAD

**Discussion by offray** [jgm/djot#289] [Ideas]
In GFM and other Markdown variants, the usual marker for checklist is binary `[ ]`: undone / `[x]`: done. But what about task with progressive completion? Emacs/OrgModo and NeoVim/Norg, support progressive checkmarks for task completion. I propose something like:

```
- [ ] undone
- [/] 1/4 done
- [-] Half done
- [\] 3/4 done.
- [x] Done
```

Which is compatible with binary tasks, while gives a visual sense of progression for the ones who aren't. If more granular advance needs to be indicated, something like `[9/10]` (nine out of then points/tasks) can be introduced.

**Comment by bdarcus** [comment]
I've mentioned that previously. But with your suggestion, there's no way to indicate a generic "in progress"?

**Reply by offray** [reply]
We could introduce new symbols for states such states:

- `[*]` in a process "spinning wheel".
-  `[=]` paused (remains equal).
-  `[.]` stopped (full stop).

I think that makes the symbols for progressive advancement and process status sufficiently mnemonic and visually distinctive.

**Reply by dereuromark** [reply]
Interesting.
I had another idea from Org-Mode:
https://github.com/php-collective/djot-php/pull/9

But it could potentially include `/` half, so any "partial done".
I dont think 25/75 percent are that important in these scenarios, right? mainly no, yes and sth in between as 3 states.

**Reply by offray** [reply]
I also like the idea of using `[/]` as an indicator of partial advancement, but I would like to use it as "clocks" introduced in table top role playing games like Blades in the Dark, which give more granular control on task progress while still being compatible with current task list, as showed in the beginning of the thread. For other discussions extending on this answer, I have created its own thread on [Format evolution: feature completeness versus extensibility](https://github.com/jgm/djot/discussions/358)

**Reply by dereuromark** [reply]
Yeah, I saw that above.
It actually made me think again now - maybe the standard should mainly be about a single char (or space) inside, and the parsing libraries define their own meaning or translation of it.
So I closed my PR in favor of this idea: https://github.com/php-collective/djot-php/pull/14
Simply allowing to store the single char makes customization possible.

So your's would be translated to maybe

  | Marker | Icon | Progress |
  |--------|------|----------|
  |        |  ○   | 0%       |
  |   /    |  ◔   | 25%      |
  |   -    |  ◑   | 50%      |
  |   \    |  ◕   | 75%      |
  |   x    |  ●   | 100%     |

and mine using CSS to some emojis etc.

But the point is: This is then really custom to the business case or the parser lib in use, and maybe not so much about the standard. As it is hard to define a common set here depending on the many use cases available.

**Reply by offray** [reply]
Your idea for a customizable parser seems pretty on point with the discussion on extensibility vs completeness, linked above. 

I use the Harvey Balls plugin for TiddlyWiki to indicate partial progress of activities, being them personal projects or role playing games and having a quick textual/visual notation for them, helps a lot with readability.

I second your proposal.

Cheers,
